source/speech/speech.py

```diff
@@ -3059,3 +3063,38 @@ def clearTypedWordBuffer() -> None:
 	complete the word (such as a focus change or choosing to move the caret).
 	"""
 	_curWordChars.clear()
+
+
+def _sapi4DeprecationWarning(synth: SynthDriver, audioOutputDevice: str, isFallback: bool):
+	"""A synthChanged event handler to alert the user about the deprecation of SAPI4."""
+
+	def setShown():
+		setattr(synth, "_hasWarningBeenShown", True)
+		synth.saveSettings()
+
+	def impl():
+		MessageDialog(
+			parent=None,
+			message=_(
+				# Translators: Message warning users that SAPI4 is deprecated.
+				"Microsoft Speech API version 4 is obsolete. "
+				"Using this speech synthesizer may pose a security risk. "
+				"This synthesizer driver will be removed in NVDA 2026.1. "
+				"You are strongly encouraged to choose a more modern speech synthesizer.",
```

Comment:
I wonder, could we add an extra button that either:

1. Shows the `Supported Speech Synthesizers` user guide section in a browseable message
2. Opens the user guide in the browser with that particular section

alternatively:

```suggestion
				"You are strongly encouraged to choose a more modern speech synthesizer."
				"Consult the Supported Speech Synthesizers section in the user guide for suggestions.",
```